### ISSUE TITLE
Line number does not consider context

### ISSUE THREAD

**Issue by jgebhardt** [fgnass/diffparser#1]
Input:
```
@@ -36,6 +36,7 @@ export type SomeContext = {
   foo: bar,
 };

+import newdep from 'newdep';
 import {bla} from 'bla';
 import {qwe} from 'qwe';
 import {ertyu} from 'ertyu';
```

Output: 
```
...
{
      "content": "@@ -36,6 +36,7 @@ export type SomeContext = {",
      "changes": [
        {
          "type": "normal",
          "normal": true,
          "position": 1,
          "content": "   foo: bar,"
        },
        {
          "type": "normal",
          "normal": true,
          "position": 2,
          "content": " };"
        },
        {
          "type": "add",
          "add": true,
          "ln": 36,
          "position": 3,
          "content": "+import newdep from 'newdep';"
        },
        {
          "type": "normal",
          "normal": true,
          "position": 4,
          "content": " import {bla} from 'bla';"
        },
        {
          "type": "normal",
          "normal": true,
          "position": 5,
          "content": " import {qwe} from 'qwe';"
        },
        {
          "type": "normal",
          "normal": true,
          "position": 6,
          "content": " import {ertyu} from 'ertyu';"
        }
      ],
      "oldStart": 36,
      "oldLines": 6,
      "newStart": 36,
      "newLines": 7
    },
...
```

There are 2 issues here:
1) The `ln` property is only present for changed lines (`type` === `add`/`del`), but not in lines of `type` `normal`.
2) When the `ln` property is present, it does not consider the number of context lines preceding it, rendering its value inaccurate. In the above example, the line marked with `ln: 36` should really be `ln: 38`.

Ideally, `parse` would return a consistent format for each changed line, including the `ln` number, the calculation for which should also consider the extent of the surrounding context.

**Comment by fgnass** [comment]
Thanks for spotting this, you're right the line numbering is broken.

In theory lines of type `normal` should have a `ln1` and `ln2` property (since their position can be different in both files).

I think in order to fix this the ternary expressions in https://github.com/fgnass/diffparser/blob/master/src/index.js#L108-L109 need to be the other way round.

For consistency _additions_ cold also have a `ln2` property and _deletions_ `ln1`, I think this would make sense.

BTW (without looking at it in depth) shouldn't `+import newdep from 'newdep';` in your example be reported as line 40?

And shouldn't `export type SomeContext = {` be a type `normal` block too with `position: 0, ln1: 36, ln2: 36`?

**Comment by fgnass** [comment]
Fixed in 2.0.0
